src/button-card.ts

```diff
@@ -0,0 +1,569 @@
+class ButtonCard extends LitElement {
+  @property() public hass?: HomeAssistant;
+
+  @property() private config?: ButtonCardConfig;
+
+  static get styles(): CSSResult {
+    return css`
+        ha-card {
+          cursor: pointer;
+          overflow: hidden;
+        }
+        ha-card.disabled {
+          pointer-events: none;
+          cursor: default;
+        }
+        ha-icon {
+          display: inline-block;
+          margin: auto;
+        }
+        div.button-card-main {
+          padding: 4% 0px;
+          text-transform: none;
+          font-weight: 400;
+          font-size: 1.2rem;
+          align-items: center;
+          text-align: center;
+          letter-spacing: normal;
+          width: 100%;
+        }
+        div.divTable{
+          display: table;
+          overflow: auto;
+          table-layout: fixed;
+          width: 100%;
+        }
+        div.divTableBody {
+          display: table-row-group;
+        }
+        div.divTableRow {
+          display: table-row;
+        }
+        .divTableCell {
+          display: table-cell;
+          vertical-align: middle;
+        }
+        div {
+          text-overflow: ellipsis;
+          white-space: nowrap;
+          overflow: hidden;
+          min-width: 100%;
+        }
+        div.button-card-background-color {
+          border-bottom-left-radius: 2px;
+          border-bottom-right-radius: 2px;
+          border-top-left-radius: 2px;
+          border-top-right-radius: 2px;
+        }
+        @keyframes blink{
+          0%{opacity:0;}
+          50%{opacity:1;}
+          100%{opacity:0;}
+        }
+        @-webkit-keyframes rotating /* Safari and Chrome */ {
+          from {
+            -webkit-transform: rotate(0deg);
+            -o-transform: rotate(0deg);
+            transform: rotate(0deg);
+          }
+          to {
+            -webkit-transform: rotate(360deg);
+            -o-transform: rotate(360deg);
+            transform: rotate(360deg);
+          }
+        }
+        @keyframes rotating {
+          from {
+            -ms-transform: rotate(0deg);
+            -moz-transform: rotate(0deg);
+            -webkit-transform: rotate(0deg);
+            -o-transform: rotate(0deg);
+            transform: rotate(0deg);
+          }
+          to {
+            -ms-transform: rotate(360deg);
+            -moz-transform: rotate(360deg);
+            -webkit-transform: rotate(360deg);
+            -o-transform: rotate(360deg);
+            transform: rotate(360deg);
+          }
+        }
+        .rotating {
+          -webkit-animation: rotating 2s linear infinite;
+          -moz-animation: rotating 2s linear infinite;
+          -ms-animation: rotating 2s linear infinite;
+          -o-animation: rotating 2s linear infinite;
+          animation: rotating 2s linear infinite;
+        }
+      `;
+  }
+
+  protected render(): TemplateResult | void {
+    let state: HassEntity | null = null;
```

Comment:
```suggestion
    let state: HassEntity | undefined = undefined;
```

Reply:
Actually we can do better and with the guard below, this gets cleaner as the linter won't complain about config or hass being undefined

```suggestion
    const state = this.config.entity ? this.hass.states[this.config.entity] : undefined;
```